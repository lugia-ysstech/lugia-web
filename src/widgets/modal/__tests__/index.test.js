/**
 *
 * create by guorg
 *
 * @flow
 *
 */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import chai from 'chai';
import Adapter from 'enzyme-adapter-react-16';
import { delay } from '@lugia/react-test-utils';

import ModalDemo from '../demo';
import Modal from '../index';

const { expect: exp } = chai;
Enzyme.configure({ adapter: new Adapter() });

describe('Modal', () => {
  const getState = (target: any): Object => {
    const state = target
      .children()
      .at(0)
      .instance().state;
    return state;
  };
  it('css', () => {
    const Target = <ModalDemo visible={true} mountBody={false} />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });
  it('Modal state: visible and opening and closing', async () => {
    const target = mount(<Modal visible={true} title="这是标题！" />);
    const state = getState(target);
    expect(state.visible).toBe(true);
    expect(state.opening).toBe(true);
    target
      .find('ButtonWrap')
      .at(0)
      .simulate('click');
    expect(getState(target).closing).toBe(false);
    await delay(500);
    expect(getState(target).closing).toBe(false);
    expect(getState(target).visible).toBe(true);
    target.setProps({ visible: false });
    expect(getState(target).opening).toBe(false);
  });
  it('Modal isInprops', () => {
    const target = mount(<Modal visible={true} title="这是标题！" />);
    const com = target
      .children()
      .at(0)
      .instance();
    expect(com.isInprops('visible')).toBe(true);
    expect(com.isInprops('footer')).toBe(false);
  });
});
