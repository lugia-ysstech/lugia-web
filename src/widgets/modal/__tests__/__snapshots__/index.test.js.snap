// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal css 1`] = `
.c0 {
  display: none;
  font-size: 1.4rem;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 99999;
}

.c1 {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0,0,0,0.65);
  z-index: 99999;
}

.c2 {
  position: fixed;
  left: 50%;
  top: 100px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 99999;
}

.c3 {
  box-sizing: border-box;
  font-size: 1.4rem;
  outline: none;
}

.c4 {
  position: relative;
  border: 0;
  border-radius: 0.4rem;
  box-shadow: 0 0.4rem 1.2rem rgba(0,0,0,0.15);
  min-width: 52rem;
  box-shadow: 0px 0px 0px 0px;
  background-color: #fff;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #ccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #ccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #ccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #ccc;
  border-top-left-radius: 0.8rem;
  border-top-right-radius: 0.8rem;
  border-bottom-right-radius: 0.8rem;
  border-bottom-left-radius: 0.8rem;
}

.c5 {
  position: absolute;
  right: 0;
  top: 0;
  width: 6.4rem;
  height: 6.4rem;
  cursor: pointer;
  text-align: center;
  line-height: 6.4rem;
}

.c7 {
  border-radius: 0.4rem 0.4rem 0 0;
  color: #363B40;
  font-size: 1.6rem;
  font-weight: 500;
  padding-top: 0rem;
  padding-bottom: 1.6rem;
  padding-left: 0rem;
  padding-right: 0rem;
}

.c8 {
  word-wrap: break-word;
  color: #646A6E;
  font-size: 1.4rem;
  padding-top: 0rem;
  padding-bottom: 0rem;
  padding-left: 0rem;
  padding-right: 0rem;
}

.c9 {
  padding-top: 2.2rem;
  border-radius: 0 0 4px 4px;
}

.c9 > button {
  margin-left: 1.4rem;
}

.c9 > button:first-child {
  margin-left: 0;
}

.c10 {
  display: inline-block;
  margin-bottom: 0;
  box-sizing: border-box;
  touch-action: manipulation;
  cursor: pointer;
  white-space: nowrap;
  font-family: 'Trebuchet MS',Arial,Helvetica,sans-serif;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  text-transform: none;
  outline: 0;
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  background-color: white;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #d0d3da;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #d0d3da;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #d0d3da;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #d0d3da;
}

.c12 {
  font-size: 1.4rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.c6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
    style={Object {}}
  />
  <div
    className="c2"
  >
    <div
      className="c3"
      style={Object {}}
      tabIndex="1"
    >
      <div
        className="c4"
        style={
          Object {
            "paddingBottom": "3rem",
            "paddingLeft": "3rem",
            "paddingRight": "3rem",
            "paddingTop": "3rem",
          }
        }
      >
        <div
          className="c5"
          onClick={[Function]}
        >
          <i
            className="lugia-icon-reminder_close   c6"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={
              Object {
                "color": "#9EA4AE",
                "fontSize": "1.4rem",
              }
            }
          />
        </div>
        <div
          className="c7"
          style={Object {}}
          title={true}
        />
        <div
          className="c8"
          style={Object {}}
        />
        <div
          className="c9"
        >
          <button
            className="c10"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            size="default"
            style={
              Object {
                "backgroundColor": "#4d63ff",
                "borderBottomLeftRadius": "0.4rem",
                "borderBottomRightRadius": "0.4rem",
                "borderBottomWidth": 0,
                "borderLeftWidth": 0,
                "borderRightWidth": 0,
                "borderTopLeftRadius": "0.4rem",
                "borderTopRightRadius": "0.4rem",
                "borderTopWidth": 0,
                "height": "3.2rem",
                "paddingBottom": "0rem",
                "paddingLeft": "1.8rem",
                "paddingRight": "1.8rem",
                "paddingTop": "0rem",
              }
            }
            type="primary"
          >
            <span
              className="c11"
            >
              <span
                className="c12"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": "1.4rem",
                  }
                }
              >
                确定
              </span>
            </span>
          </button>
          <button
            className="c10"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            size="default"
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#d0d3da",
                "borderBottomLeftRadius": "0.4rem",
                "borderBottomRightRadius": "0.4rem",
                "borderBottomStyle": "solid",
                "borderBottomWidth": "0.1rem",
                "borderLeftColor": "#d0d3da",
                "borderLeftStyle": "solid",
                "borderLeftWidth": "0.1rem",
                "borderRightColor": "#d0d3da",
                "borderRightStyle": "solid",
                "borderRightWidth": "0.1rem",
                "borderTopColor": "#d0d3da",
                "borderTopLeftRadius": "0.4rem",
                "borderTopRightRadius": "0.4rem",
                "borderTopStyle": "solid",
                "borderTopWidth": "0.1rem",
                "height": "3.2rem",
                "paddingBottom": "0rem",
                "paddingLeft": "1.8rem",
                "paddingRight": "1.8rem",
                "paddingTop": "0rem",
              }
            }
          >
            <span
              className="c11"
            >
              <span
                className="c12"
                style={
                  Object {
                    "color": "#363B40",
                    "fontSize": "1.4rem",
                  }
                }
              >
                取消
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
